### PR TITLE
chore: update JNBullet to 1.0.4

### PR DESCRIPTION
**Pull request by skaldarnar**
### Contains

With JNBullet v1.0.4(-SNAPSHOT) we bundle natives for both Intel and Apple Silicon Macs. 

Nothing should change for Linux and Windows.

Related to MovingBlocks/JNBullet#15.

Contributes to #5055.

### How to test

Just checkout this PR and run the game from source.

- on Mac, at least the game menu should show up (there are more issues with OpenGL on M1, at least)
- on Linux and Windows, the game should run as before

### Outstanding before merging

- [x] relase JNBullet 1.0.4 if the SNAPSHOT version tests out fine
- [x] consume the non-SNAPSHOT version here

### CODE REVIEW

**Comment by skaldarnar** [comment]
✅  tests out fine on M1 Mac (_I can build and start the game into the main menu, running the game crashes due to some OpenGL error. I think the Vulkan workstream had made some progress in that direction.)

File: build.gradle.kts [hunk 1]
```diff
@@ -98,7 +98,7 @@ dependencies {
     natives(group = "org.terasology.jnlua", name = "jnlua_natives", version = "0.1.0-SNAPSHOT", ext = "zip")
 
     // Natives for JNBullet
-    natives(group = "org.terasology.jnbullet", name = "JNBullet", version = "1.0.2", ext = "zip")
+    natives(group = "org.terasology.jnbullet", name = "JNBullet", version = "1.0.4-SNAPSHOT", ext = "zip")
```
Comment (soloturn) [inline build.gradle.kts]: it would be better to have release versions not snapshot versions?

**Comment by jdrueckert** [comment]
Tests out fine on Linux and Windows

**Comment by BenjaminAmos** [comment]
Everything should be ready for releasing JNBullet v1.0.4. It just needs a final non-snapshot release to be published now.

**Review by soloturn — APPROVED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (jdrueckert) [reply build.gradle.kts]: that's exactly why the PR wasn't merged before and why I updated it with https://github.com/MovingBlocks/Terasology/pull/5225/commits/496bdafa860c222e62f3bb96487156975f24fcec :thinking:

**Pull request merged**